Sources/SmartInsights/UpdateAvailableInsight.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  UpdateAvailableInsight.swift
+//  Diagnostics
+//
+//  Created by Antoine van der Lee on 09/02/2022.
+//  Copyright © 2019 WeTransfer. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+/// Uses the bundle identifier to fetch latest version information and provides insights into whether
+/// an app update is available.
+struct UpdateAvailableInsight: SmartInsightProviding {
+    
+    let name = "Update available"
+    let result: InsightResult
+    
+    private var urlSessionAppMetadataPublisher: (URL) -> AnyPublisher<AppMetadataResults, Error> = { url in
+        URLSession.shared
+            .dataTaskPublisher(for: url)
+            .map(\.data)
+            .decode(type: AppMetadataResults.self, decoder: JSONDecoder())
+            .eraseToAnyPublisher()
+    }
+    
```

Comment:
<!--
  0 failure: 
  1 warning:  Lines should not ...
  
  
  DangerID: danger-id-Danger;
  File: Sources/SmartInsights/UpdateAvailableInsight.swift;
  Line: 26;
-->

- :warning: Lines should not have trailing whitespace. (`trailing_whitespace`)